### PR TITLE
Hub - add separate options for sending / receiving line termination characters

### PR DESCRIPTION
**Pull request by icklesteve**
Enables independent setting of sending and receiving line termination characters.

### CODE REVIEW

File: src/org/openlcb/hub/Hub.java [hunk 1]
```diff
@@ -37,20 +37,40 @@ public class Hub {
     private final static Logger logger = Logger.getLogger(Hub.class.getName());
     public final static int DEFAULT_PORT = 12021;
     final static int CAPACITY = 20;  // not too long, to reduce delay
-    private final boolean lineEndings;
+    private final boolean sendLineEndings;
+    private final boolean receiveLineEndings;
```
Comment (balazsracz) [inline src/org/openlcb/hub/Hub.java:41]: let's call this `requireReceivedLineEndings` or `requireIncomingLineEndings`

**Review by balazsracz — APPROVED** [review]

File: src/org/openlcb/hub/Hub.java (hunk 1 above)
Comment (icklesteve) [reply src/org/openlcb/hub/Hub.java:41]: Thanks, will update.

File: src/org/openlcb/hub/Hub.java [hunk 2]
```diff
@@ -37,20 +37,41 @@ public class Hub {
...
-    private final boolean lineEndings;
+    private final boolean sendLineEndings;
+    private final boolean requireIncomingLineEndings;
     private boolean disposed = false;
 
+    /**
+     * Constructs a new instance using default values.
+     * Port - 12021
+     * Send line endings - true
+     * Require incoming line endings - true
```
Comment (balazsracz) [inline src/org/openlcb/hub/Hub.java:48]: I liked it better in the previous iteration of the PR when we had require incoming line endings == false as default.
This provides the most compatibility.

I understand that this is a change in behavior from the previous code, but I believe wider compatibility is better, so I will approve this change in behavior.

**Review by balazsracz — COMMENTED** [review]
thanks, this looks easier to read.

File: src/org/openlcb/hub/Hub.java (hunk 2 above)
Comment (icklesteve) [reply src/org/openlcb/hub/Hub.java:48]: No problem, will change the default back to the new version.

**Review by bobjacobsen — APPROVED** [review]
Looks good to me.

**Pull request merged**